eli5/sklearn/text.py

```diff
@@ -13,10 +13,11 @@
 
 
 def get_weighted_spans(doc, vec, feature_weights):
-    # type: (Any, Any, FeatureWeights) -> Optional[WeightedSpans]
+    # type: (Any, Any, Union[FeatureWeights, None]) -> Optional[WeightedSpans]
```

Comment:
it seems this function requires FeatureWeights to be not None, so maybe it makes sense to keep type signature the same, but move assert to the caller code

Reply:
Moved the assert to `add_weighted_spans`. Good one. Strange that https://github.com/TeamHG-Memex/eli5/search?q=get_weighted_spans did not show `add_weighted_spans`'s call to `get_weighted_spans`.

https://github.com/TeamHG-Memex/eli5/pull/315/commits/913d415e6bb3c2b5eeedefd164ff0b43bc3e7487